src/main/java/com/artipie/rpm/Repomd.java

```diff
@@ -160,10 +155,26 @@ private CompletableSource performUpdate(
                     file
                 )
             )
-            .flatMap(file -> act.update(file).andThen(Single.just(file)))
-            .zipWith(
-                Single.fromCallable(() -> Files.createTempFile(type, ".xml.gz")),
-                (src, gzip) -> Repomd.gzip(src, gzip).andThen(
+            .flatMap(file -> act.update(file).toSingleDefault(file))
+            .flatMap(file -> this.buildRepoXml(type, file))
+            .flatMapCompletable(new Update(repomd)::apply)
+            .andThen(this.saveRepo(repomd));
+    }
+
+    /**
+     * Build repo xml by preparing its content.
+     *
+     * @param type The file type
+     * @param src Source metadata file
+     * @return Completion or error signal
+     * @todo #60:30min Transform this method and all its dependent (in
+     *  particular: saveGzip, repoXml, gzip, etc) in a self-sufficient
+     *  class that extends Single, and add tests for it.
```

Comment:
@victornoel why should those classes extend `Single`? They return `Completable`.

Reply:
@vatavuk it should be only ONE class that wraps `buildRepoXml` and hide its dependent. `buildRepoXml` returns a `Single` because it generates a new object (while `Completable` is only a procedure that returns nothing).

Reply:
@victornoel ok, I've understood you differently, to have multiple classes. ok we can merge